### PR TITLE
Tests checking documentation.

### PR DESCRIPTION
**Pull request by JohanLarsson**
If a new analyzer is added with no documentation the test MissingDocs fails and creates a stub doc Id.md.generated.
Remove the .generated suffix and add it to the solution folder /docs
Not sure if the extra step with .generated suffix solves a problem but that is how I originally wrote it for some reason.

Currently the analyzers do not have Description, the stub docs have a `## Description` with an empty value and the test for description is set to [Explicit]. If no description is by design we should remove the test and the noise from the stub docs.

#96

We create help links in the analyzers so that clicking on the id in the VS error list navigates to the documentation. I can PR that later.

The code in this PR is not awesome, @jnm2 said it was fascinating, he is a polite man :)

### CODE REVIEW

File: documentation/NUnit1001.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+| Topic    | Value
+| :--      | :--
+| Id       | NUnit1001
+| Severity | Error
+| Enabled  | True
+| Category | Structure
+| Code     | [TestCaseUsageAnalyzer](https://github.com/nunit/nunit.analyzers/blob/master/src/nunit.analyzers/TestCaseUsage/TestCaseUsageAnalyzer.cs)
+
+
+## Description
```
Comment (JohanLarsson) [inline documentation/NUnit1001.md]: No description provided in the descriptors. I included this header with empty value in the generated stubs.

File: src/nunit.analyzers.tests/DocumentationTests.cs [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        {
+            var expected = $"## {descriptorInfo.Descriptor.Title}";
+            var actual = descriptorInfo.DocumentationFile.AllLines
+                                       .Skip(1)
+                                       .First()
+                                       .Replace("`", string.Empty);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Explicit("No descriptions. Remove this test and update doc stubs if we don't want descriptions.")]
```
Comment (JohanLarsson) [inline src/nunit.analyzers.tests/DocumentationTests.cs]: This test should be removed or `[Explicit]` should be removed depending on if we want description.

File: documentation/NUnit1001.md (hunk 1 above)
Comment (mikkelbu) [reply documentation/NUnit1001.md]: I think it is fine. We should have descriptions in this project

File: src/nunit.analyzers.tests/DocumentationTests.cs (hunk 2 above)
Comment (mikkelbu) [reply src/nunit.analyzers.tests/DocumentationTests.cs]: We should add descriptions and then remove `[Explicit]`.

**Comment by mikkelbu** [comment]
Thank you very much for the contribution @JohanLarsson. This will save us a lot of work. Would you mind if I add a commit to this PR that reformats the code in `DocumentationTests.cs`, so that it follows the rest of the project (I think that this would be much faster than writing a lot of nit-pick comments about syntax). Alternative I can also merge this PR as it is, and then subsequently change the code a little.

Ps. I think you are being too hard on the code :). I was fearing something much more "interesting" from your descriptions :D.

**Comment by JohanLarsson** [comment]
Of course I don't mind, I copy pasted, forgot about project conventions, sorry about that.

**Comment by jnm2** [comment]
@JohanLarsson Oh dear, I meant "fascinating" more literally... as in, I'm interested in  this kind of workflow, the idea of autogenerating and maintaining docs via the test suite. I'm so sorry for giving the wrong impression!

**Pull request merged**